src/Layout/reducers.js

```diff
@@ -1,13 +1,14 @@
 const bar = (state = {}, action) => {
   switch (action.type) {
-  case 'ACTIVE_INIT': {
-    return { ...state, active: window.location.pathname };
-  }
   case 'ACTIVE_CHANGED': {
     return { ...state, active: action.data };
   }
   default:
-    return state;
+    // Keep active in sync with URL for navigation updates outside of NavBar
+    if (state.active === window.location.pathname) {
```

Comment:
I think this violates one of the three principles of redux; that a reducer be a pure function (looking at the diff it seems like we were already doing that, but maybe now's a good time to re-think it)

https://redux.js.org/understanding/thinking-in-redux/three-principles

Ideally, a reducer run multiple times with the same parameters yields the same results. I think the action(s) that trigger navigation changes or the components that render changes related to the current path should probably be where we compare the current path.

Is the problem that we have some code where we're navigating without using react-router? And we can't reliably determine what the current nav item is?

Reply:
the problem is that the NavBar does not update when changes are made outside of it. 
there are pages that use `this.props.history.push('/workspace');` that NavBar then does not update from 

i am new to react and react router so any suggestions on how to properly solve this i am open to

Reply:
I didn't test but maybe we can add a `ComponentDidUpdate` call inside `NavBar`, check pathname in there and dispatch action if needed. 🤔

Reply:
I didn't test but maybe we can add a `ComponentDidUpdate` call inside `NavBar`, check pathname in there and dispatch action if needed. 🤔 

But I would hightly suggest to investigate the possibility for a more elegant fix. The current logic flow of `NavBar` which contains a lost of customized `props`, `states` and `actions` is really concerning for me. To my knowledge this kind of work should be able to handeled by `react-router` completely. But this `NavBar` code is 3 years old, whoe knows what happened back then 🤷‍♂️ 

Here is a quick demo I put together using `NavLink` https://codesandbox.io/s/react-router-navlink-meeg2 , it can hightlight the active UI component no matter if a link is clicked, a button is clicked, or directly update URL in address bar. If we can refactor our `NavBar` component using something similar to this, I think it will be a much better solution.

Reply:
💯 ^

Reply:
will redo this all, thanks for the input, have wasted too much time trying to "fix" it